Add minimumMeasurementCountThreshold to cdn preference schema

danieltibbetts-digicert opened the pull request.

danieltibbetts-digicert committed:
Add minimumMeasurementCountThreshold to cdn preference

anuplingraraj commented on provider/preference_resource.go:
```diff
@@ -216,6 +223,33 @@ func (r *preferenceResource) Schema(_ context.Context, _ resource.SchemaRequest,
+					},
+					"continents": schema.MapNestedAttribute{
+						Description: "Continent-specific minimum measurement counts",
+						Required:    true,
```
this is optional. Check openAPI spec to confirm once.

danieltibbetts-digicert replied:
Both the world and continents fields are required, according to the spec

anuplingraraj commented on examples/basic/main.tf:
```diff
@@ -157,4 +157,16 @@ resource "multicdn_preference_config" "example_website_preferences" {
+
+  minimum_measurement_count_threshold = {
+    world = 5
+    continents = {
```
I don't see continents being populated in mongo backend. Please check once if the entire object is being passed. By default, backend service will create just world threshold if nothing is passed so that's what happening now.

danieltibbetts-digicert replied:
Fixed

danieltibbetts-digicert committed:
Fix + code review changes

anuplingraraj committed:
Make preference config schema attributes optional

Changed several attributes in the preferenceResource schema from required to optional, including 'world', 'continents', 'default', and 'countries'. Updated example Terraform configuration with real API credentials, a new base URL, and added 'MX' to the NA countries list.

Junaid-Ahm commented on clients/preferenceclient/operations_test.go:
Should add tests for create and update operations to also handle the newly added field.

Junaid-Ahm replied:
Currently, a test is only added for the GET operation.

Junaid-Ahm commented on provider/preference_resource.go:
No tests were added for co-located file: `preference_resource_test.go`

Tests should be added to validate all schema operations: GET, Create, Update, Import are working. 

See existing examples and just add new test cases (to existing test suites) to cover the newly added fields.
